src/Service.Host/client/src/components/SerialNumber.js

```diff
@@ -0,0 +1,387 @@
+import React, { Fragment, useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Grid, Button } from '@material-ui/core';
+import {
+    SaveBackCancelButtons,
+    InputField,
+    Loading,
+    Title,
+    ErrorCard,
+    SnackbarMessage,
+    Dropdown,
+    useSearch,
+    AutoComplete
+} from '@linn-it/linn-form-components-library';
+import Page from '../containers/Page';
+import { getSernosNote } from '../selectors/sernosNotesSelectors';
+
+function SerialNumber({
+    errorMessage,
+    editStatus,
+    history,
+    loading,
+    item,
+    salesArticleSernosDetails,
+    snackbarVisible,
+    addItem,
+    salesArticlesSearchResults,
+    setEditStatus,
+    sernosNotes,
+    sernosTransactions,
+    sernosTransactionsLoading,
+    setSnackbarVisible,
+    fetchSalesArticleSernosDetails,
+    fetchSalesArticles,
+    salesArticlesLoading,
+    clearSerialNumber,
+    clearSalesArticleSernosDetails
+}) {
+    const [serialNumber, setSerialNumber] = useState({});
+    const [prevSerialNumber, setPrevSerialNumber] = useState({});
+    const [sernosTransactionsList, setSernosTransactionsList] = useState(['']);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [salesArticles, setSalesArticles] = useState([]);
+
+    const savedFetchSalesArticleSernosDetails = useRef(null);
+
+    useSearch(fetchSalesArticles, searchTerm);
+
+    useEffect(() => {
+        savedFetchSalesArticleSernosDetails.current = fetchSalesArticleSernosDetails;
+    }, [fetchSalesArticleSernosDetails]);
+
+    useEffect(() => {
+        setSalesArticles(() =>
+            salesArticlesSearchResults.map(s => ({
+                ...s,
+                label: `${s.articleNumber}: ${s.description}`
+            }))
+        );
+    }, [salesArticlesSearchResults]);
+
+    useEffect(() => {
+        if (item === null) {
+            setSerialNumber({});
+            setPrevSerialNumber(null);
+            setSalesArticles([{}]);
+        } else {
+            const sernos = item[0];
+
+            setSerialNumber(s => ({
```

Comment:
This is so that I can update the existing serial number (done with the use state hook) without having the effect triggered each time the serial number object updates. The effect will only run if item or prevSerialNumber change.